salt/modules/reg.py

```diff
@@ -69,104 +69,76 @@ def __virtual__():
     if not HAS_WINDOWS_MODULES:
         return (False, 'reg execution module failed to load: '
                        'One of the following libraries did not load: '
-                       + '_winreg, win32gui, win32con, win32api')
+                       + 'win32gui, win32con, win32api')
 
     return __virtualname__
 
 
-# winreg in python 2 is hard coded to use codex 'mbcs', which uses
-# encoding that the user has assign. The function _unicode_to_mbcs
-# and _unicode_to_mbcs help with this.
-
-
-def _unicode_to_mbcs(instr):
+def _to_mbcs(vdata):
     '''
-    Converts unicode to to current users character encoding.
+    Converts unicode to to current users character encoding. Use this for values
+    returned by reg functions
     '''
-    if isinstance(instr, six.text_type):
-        # unicode to windows utf8
-        return instr.encode('mbcs')
-    else:
-        # Assume its byte str or not a str/unicode
-        return instr
+    return salt.utils.to_unicode(vdata, 'mbcs')
 
 
-def _mbcs_to_unicode(instr):
+def _to_unicode(vdata):
     '''
-    Converts from current users character encoding to unicode.
-    When instr has a value of None, the return value of the function
-    will also be None.
+    Converts from current users character encoding to unicode. Use this for
+    parameters being pass to reg functions
     '''
-    if instr is None or isinstance(instr, six.text_type):
-        return instr
-    else:
-        return six.text_type(instr, 'mbcs')
-
-
-def _mbcs_to_unicode_wrap(obj, vtype):
-    '''
-    Wraps _mbcs_to_unicode for use with registry vdata
-    '''
-    if vtype == 'REG_BINARY':
-        # We should be able to leave it alone if the user has passed binary data in yaml with
-        # binary !!
-        # In python < 3 this should have type str and in python 3+ this should be a byte array
-        return obj
-    if isinstance(obj, list):
-        return [_mbcs_to_unicode(x) for x in obj]
-    elif isinstance(obj, six.integer_types):
-        return obj
-    else:
-        return _mbcs_to_unicode(obj)
+    return salt.utils.to_unicode(vdata, 'utf-8')
```

Comment:
Why is this 'utf-8' instead of 'mbcs'

Reply:
'mbcs' wasn't translating unicode characters correctly in my test environment. We're going straight utf-8 into the pywin32 functions and sending any output through the `_to_mbcs` function. From what I was able to gather, `mbcs` just means the default encoding for the system rather than any specific type of encoding.

Reply:
mbcs is the encode for the language in use. The assumption is the data in the registry is this.  So if it was a Japanese version of windows mbcs..   Its also doco some where that this is what winreg uses to encode.    PY3 fixes all this mess.    It also maybe dependant on the version of pywin.

Reply:
Their are some tests for reg and unicode.

Reply:
The data in the registry is unicode or ascii (https://msdn.microsoft.com/en-us/library/windows/desktop/ms724884(v=vs.85).aspx). The functions we're using to put stuff in the registry accept unicode as input. We're using 'utf-8' because it seems to handle most cases and works well. We could try utf-16, but I didn't do any testing with it.

We're converting the return data from those functions (`reg.read_value`) to whatever the locale encoding is. That's handled by the `_to_mbcd` function.

Reply:
Use regedit to place this in the registry  发送时间   I think it means date
In python 2 you will get ???? in python 3 you will get \u53d1\u9001\u65f6\u95f4

Reply:
This seems like a valid concern. @twangboy were you able to look at this?

Reply:
This has always been an issue in Py2.